File: src/main/java/ru/dkolmogortsev/task/search/SearchFields.java
package ru.dkolmogortsev.task.search;

import ru.dkolmogortsev.task.Task;

import java.util.function.Function;

/**
 * Created by dkolmogortsev on 2/18/17.
 */
public enum SearchFields {
    TASKNAME(Task::getTaskName), DESCRIPTION(Task::getDescription);

    public Function<Task, String> getMapper() {
        return mapper;
    }

    private Function<Task, String > mapper;

    SearchFields(Function<Task,String> toStringMapper){
        this.mapper = toStringMapper;
    }

}

File: src/main/kotlin/ru/dkolmogortsev/task/search/SearchFields.kt
package ru.dkolmogortsev.task.search

import ru.dkolmogortsev.task.Task

/**
 * Created by dkolmogortsev on 2/18/17.
 */
enum class SearchFields
{
    TASK_NAME({ "${it.taskName}(${it.description})" }, "taskName"),
    DESCRIPTION({ "${it.description}:(${it.taskName})" }, "description");

    val mapper: (Task) -> String
    private val fieldName: String

    constructor(mapper: (Task) -> String, fieldName: String)
    {
        this.mapper = mapper
        this.fieldName = fieldName
    }

    override fun toString(): String
    {
        return fieldName
    }
}
